libpod/container_api.go

```diff
@@ -524,6 +524,14 @@ func (c *Container) WaitWithInterval(waitTimeout time.Duration) (int32, error) {
 	if !c.valid {
 		return -1, define.ErrCtrRemoved
 	}
+
+	exitFile := c.exitFilePath()
+	chWait := make(chan error)
+	defer close(chWait)
+	if _, err := WaitForFile(exitFile, chWait, 0); err != nil {
+		logrus.Errorf("Error waiting for file %s: %v", exitFile, err)
+	}
+
 	err := wait.PollImmediateInfinite(waitTimeout,
```

Comment:
Do we still need this, if we're doing the waiting for the file above? I'd just as soon drop the polling wait and convert entirely to `WaitForFile()`

Reply:
tests are failing as they get stuck in WaitForFile.  I was just rewriting it to use both (in a select block).  What do you think?

Reply:
Doesn't WaitForFile already do a poll alongside the inotify? I don't think we need to keep another one here. Might want to let you set the interval on the polling in WaitForFile though.

Failures are probably the file never appearing due to `--rm` - we'll need a timeout on this.

Reply:
fixed in the last version I've pushed